File: backend/src/main/java/com/sliit/smartcampus/service/NotificationService.java
package com.sliit.smartcampus.service;

import com.sliit.smartcampus.model.Notification;
import java.util.Collection;
import java.util.List;

public interface NotificationService {
    List<Notification> getUserNotifications(Collection<String> userIds);
    List<Notification> getUnreadNotifications(Collection<String> userIds);
    long getUnreadCount(Collection<String> userIds);
    Notification markAsRead(String notificationId, Collection<String> userIds);
    void markAllAsRead(Collection<String> userIds);
    Notification createNotification(String userId, String title, String message, String type);
    void deleteNotification(String notificationId, Collection<String> userIds);
}

File: frontend/src/features/notifications/services/notificationService.js
import API from '../../../services/api';

const notificationService = {
  getNotifications: () => API.get('/notifications'),
  getUnreadNotifications: () => API.get('/notifications/unread'),
  getUnreadCount: () => API.get('/notifications/count'),
  markAsRead: (id) => API.put(`/notifications/${id}/read`),
  markAllAsRead: () => API.put('/notifications/read-all'),
  deleteNotification: (id) => API.delete(`/notifications/${id}`)
};

export default notificationService;
